src/core/lib/promise/activity.h

```diff
@@ -387,32 +387,42 @@ class FreestandingActivity : public Activity, private Wakeable {
 };
 
 // Implementation details for an Activity of an arbitrary type of promise.
-// There should exist a static function:
+// There should exist an inner template class `BoundScheduler` that provides
+// the following interface:
 // struct WakeupScheduler {
 //   template <typename ActivityType>
-//   void ScheduleWakeup(ActivityType* activity);
+//   class BoundScheduler {
+//    public:
+//     BoundScheduler(ActivityType);
```

Comment:
Should this comment state that the BoundScheduler constructor take a WakeupScheduler type instead of an ActivityType ?